alb/terraform-examples/cluster-deployment/cluster-deployment.tf

```diff
@@ -0,0 +1,175 @@
+  count = 3
+
+  provisioner "local-exec" {
+    command = "count=1 ; while [ $(curl -k -s -o /dev/null -w \"%%{http_code}\"   https://${var.avi_ctrl_mgmt_ips[count.index]}) != \"200\" ]; do echo \"Attempt $count: Waiting for Avi Controllers to be ready...\"; sleep 5 ; count=$((count+1)) ;  if [ \"$count\" = 120 ]; then echo \"ERROR: Unable to connect to Avi Controller API\" ; exit 1 ; fi ; done"
+  }
+}
+
+
+provider "avi" {
+  avi_username   = var.avi_username
+  avi_password   = "58NFaGDJm(PJH0G"
+  avi_controller = var.avi_ctrl_mgmt_ips[0]
+  avi_tenant     = "admin"
+  avi_version    = var.avi_version
+}
+
+resource "avi_cluster" "vmware_cluster" {
+  depends_on = [null_resource.wait_https_controllers]
+  name   = "cluster-0-1"
+  virtual_ip { 
+    type = "V4"
+    addr =  var.clustervip 
+   }
+  nodes {
+    ip {
+      type = "V4"
+      addr = var.avi_ctrl_mgmt_ips[0]
+    }
+    name = vsphere_virtual_machine.controller[0].name
+  }
+  nodes {
+    ip {
+      type = "V4"
+      addr = var.avi_ctrl_mgmt_ips[1]
+    }
+    name = vsphere_virtual_machine.controller[1].name
+  }
+  nodes {
+    ip {
+      type = "V4"
+      addr = var.avi_ctrl_mgmt_ips[2]
+    }
+    name = vsphere_virtual_machine.controller[2].name
+  }
+}
+
+resource "null_resource" "wait_https_cluster" {
+  depends_on = [avi_cluster.vmware_cluster]
+  count = 3
+
+  provisioner "local-exec" {
+    command = "count=1 ; while [ $(curl -k -s -o /dev/null -w \"%%{http_code}\"   https://${var.clustervip}) != \"200\" ]; do echo \"Attempt $count: Waiting for Avi Controllers to be ready...\"; sleep 5 ; count=$((count+1)) ;  if [ \"$count\" = 120 ]; then echo \"ERROR: Unable to connect to Avi Controller API\" ; exit 1 ; fi ; done"
+  }
+}
+
+
+resource "avi_backupconfiguration" "backupconfig" {
+  depends_on = [null_resource.wait_https_cluster]
+  name = "Backup-Configuration"
+  save_local = true
+  backup_passphrase = var.backup_passphrase
+}
+
+resource "avi_systemconfiguration" "avi_sysconfig" {
+  depends_on = [avi_backupconfiguration.backupconfig]
+  dns_configuration {
+    dynamic "server_list" {
+      for_each = var.system_dnsresolvers
+      content {
+        type  = "V4"
+        addr = server_list.value
+      }
+    }
+  }
+  ntp_configuration {
+    dynamic "ntp_servers" {
+      for_each = var.system_ntpservers
+      content {
+          server {
+            type  = "DNS"
+            addr = ntp_servers.value
+          }
+      }
+    }
+  }
+  email_configuration {
+    smtp_type = "SMTP_LOCAL_HOST"
+    from_email = "admin@avicontroller.net"
+    mail_server_name = "localhost"
+  }
+  global_tenant_config {
+    tenant_vrf = false
+    se_in_provider_context = true
+    tenant_access_to_provider_se = true
+  }  
+  welcome_workflow_complete = true
+}
+resource "avi_useraccount" "avi_user" {
+  depends_on = [avi_systemconfiguration.avi_sysconfig]
+  username     = var.avi_username
+  old_password = "58NFaGDJm(PJH0G"
```

Comment:
`var.avi_password`?

Reply:
This is the original password at time of deployment. I put it in the script as I felt it wasnt needed in the input variables file.

Reply:
I would place it in a `terraform.tfvars.example` file. See the PCA examples.

Reply:
Its the default password it is needed for this to connect. I dont see the problem with leaving it here. It is publicly known default password. We have it as a note when downloading the ova.

Reply:
I added it to the variables file.